### PR TITLE
Add legend to dashboard cells

### PR DESCRIPTION
**Pull request by goller**
- [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1981

### The problem
Need backend support for legends in cells

### The Solution
A new key, `legend`, has been added to all cells.  
`legend` is an object with two parameters: `type` and `orientation`.
Currently, `type` can only be "static";  `orientation` can be one of "bottom", "top", "left", "right".

```json
{
...
    "legend": {
        "type": "static",
        "orientation": "bottom"
    }
...
}
```

Cells without legends will return the an empty object like this:

```json
{
    "legend": {}
}
```

### CODE REVIEW

**Comment by alexpaxton** [comment]
Sweeeeeeeeet you da man @goller

File: kapacitor/vars.go [hunk 1]
```diff
@@ -197,6 +198,13 @@ func groupBy(q *chronograf.QueryConfig) string {
 	return "[" + strings.Join(groups, ",") + "]"
 }
 
+func idVar(q *chronograf.QueryConfig) string {
+	if len(q.GroupBy.Tags) > 0 {
+		return `name + ':{{.Group}}'`
+	}
+	return "name"
```
Comment (desa) [inline kapacitor/vars.go]: What was the motivation for this? This change could be potentially problematic, why doesn't the previous way of

```
name + ':{{.Group}}'
```

work?

File: server/cells.go [hunk 2]
```diff
@@ -28,37 +28,31 @@ type dashboardCellResponse struct {
 
 func newCellResponse(dID chronograf.DashboardID, cell chronograf.DashboardCell) dashboardCellResponse {
 	base := "/chronograf/v1/dashboards"
-	newCell := chronograf.DashboardCell{}
-	newCell.Queries = make([]chronograf.DashboardQuery, len(cell.Queries))
-	copy(newCell.Queries, cell.Queries)
+	if cell.Queries == nil {
+		cell.Queries = make([]chronograf.DashboardQuery, 0)
```
Comment (desa) [inline server/cells.go]: super nit-picky, but any reason for the preference of make to

```go
cell.Queries = []chronograf.DashboardQuery{}
```

File: server/cells.go [hunk 3]
```diff
@@ -28,37 +28,31 @@ type dashboardCellResponse struct {
...
+	}
+	if cell.CellColors == nil {
+		cell.CellColors = make([]chronograf.CellColor, 0)
```
Comment (desa) [inline server/cells.go]: ditto above. super nit-picky also

File: kapacitor/vars.go (hunk 1 above)
Comment (goller) [reply kapacitor/vars.go]: hmm... looks like @alexpaxton changed the base branch from master to feature/persistent-legend.

I think persistent-legend is behind and so A LOT more files ended up in this PR.

File: server/cells.go (hunk 2 above)
Comment (goller) [reply server/cells.go]: Fixed in 84b6cf4

File: server/cells.go (hunk 3 above)
Comment (goller) [reply server/cells.go]: Fixed in 84b6cf4

**Pull request merged**